backend/src/v2/component/launcher_v2.go

```diff
@@ -427,6 +433,46 @@ func prettyPrint(jsonStr string) string {
 
 const OutputMetadataFilepath = "/tmp/kfp_outputs/output_metadata.json"
 
+// We overwrite this as a DI mechanism for testing getLogWriter. We could make
+// it an explicit input to getLogWriter, but then it would have to be an input
+// in getLogWriter's caller as well.
+var osCreateFunc = os.Create
+
+// getLogWriter returns an io.Writer that can either be single-channel to stdout
+// or dual-channel to stdout AND a log file based on the URI of a log artifact
+// in the supplied ArtifactList. Downstream, the resulting log file gets
+// uploaded to the object store.
+func getLogWriter(artifacts map[string]*pipelinespec.ArtifactList) (writer io.Writer) {
+	logsArtifactList, ok := artifacts["logs"]
+
+	if !ok || len(logsArtifactList.Artifacts) == 0 {
+		return os.Stdout
+	}
```

Comment:
couple of questions: 

1. can we have more than one artifacts["logs"]? 
2. when can we not have logs? when we don't opt in? 

afaik we haven't made this optional so I'm wondering if we should be explicit here and ensure there is always one and only one "logs" entry? 

(lmk if I have misunderstood this one)

Reply:
AFAIK, the values are either 1 (logs are enabled) or 0 (logs are disabled and / or we're executing a run from before this feature was implemented).

That being said, I'll make the check more explicit: `!= 1` instead of `== 0`.